File: server/src/config/catchyException.ts
import { HttpException } from '@nestjs/common';
import { HTTP_STATUS_CODE } from 'src/httpStatusCode.enum';
import { ERROR_CODE } from './errorCode.enum';

export class CatchyException extends HttpException {
  constructor(message: string, statuscode: HTTP_STATUS_CODE, errorCode: ERROR_CODE) {
    super({message, errorCode}, statuscode);
  }
}

File: server/spring/src/main/java/catchytape/spring/common/exception/CatchyException.java
package catchytape.spring.common.exception;

import catchytape.spring.common.domain.ErrorCode;
import catchytape.spring.common.domain.HttpStatusCode;
import lombok.Getter;

@Getter
public class CatchyException extends Exception {
    private int statusCode;
    private int errorCode;

    public CatchyException(String statusMessage, String errorMessage) {
        super(errorMessage);
        this.statusCode = HttpStatusCode.valueOf(statusMessage).getCode();
        this.errorCode = ErrorCode.valueOf(errorMessage).getCode();
    }
}
